### PR TITLE
GEOMESA-2609 Parquet converter implementation

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Add file path to converter inference

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
note: stacks on top of #2262

**Comment by jnh5y** [comment]
Needs a rebase.:)

I'm happy to merge #2259 first if it'd help.

**Comment by elahrvivaz** [comment]
rebased

File: docs/user/convert/parquet.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+.. _parquet_converter_functions:
+
+Parquet Transform Functions
+---------------------------
+
+GeoMesa defines several Parquet-specific transform functions, in addition to the ones defined under
+:ref:`avro_converter_functions`.
+
+parquetPoint
```
Comment (jnh5y) [inline docs/user/convert/parquet.rst]: When are these needed?  The example below uses paths and existing functions.  Is this syntactical sugar?

File: geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/SimpleFeatureConverterFactory.scala [hunk 2]
```diff
@@ -31,5 +32,26 @@ trait SimpleFeatureConverterFactory {
     * @param sft simple feature type, if known ahead of time
     * @return
     */
-  def infer(is: InputStream, sft: Option[SimpleFeatureType] = None): Option[(SimpleFeatureType, Config)] = None
+  @deprecated("Use infer(InputStream, Option[String], Option[SimpleFeatureType]")
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/SimpleFeatureConverterFactory.scala]: Do you mean:
  @deprecated("Use infer(InputStream, Option[SimpleFeatureType],  Option[String]")
?

File: docs/user/convert/parquet.rst (hunk 1 above)
Comment (elahrvivaz) [reply docs/user/convert/parquet.rst]: These are for geomesa-encoded geometry columns, as written by geomesa-fs-storage-parquet. The examples below are just pulling lat/lon out of generic parquet data, so they're not needed. Our geometry columns use more complicated nested structures, that wouldn't be easy (possible?) to read using regular paths and functions.

Comment (elahrvivaz) [reply docs/user/convert/parquet.rst]: generally these will probably only show up in type-inferenced ingestion of geomesa-parquet files.

File: geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/SimpleFeatureConverterFactory.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/SimpleFeatureConverterFactory.scala]: yes, thanks

File: geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/io/SimpleFeatureReadSupport.scala [hunk 3]
```diff
@@ -181,12 +216,12 @@ object SimpleFeatureReadSupport {
     override def addBinary(value: Binary): Unit = callback.set(index, value.getBytes)
   }
 
-  class ListConverter(index: Int, binding: Seq[ObjectType], callback: Settable) extends GroupConverter {
+  class ListConverter(binding: ObjectType, index: Int, callback: Settable) extends GroupConverter {
```
Comment (jnh5y) [inline geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/io/SimpleFeatureReadSupport.scala]: Any particular reason for the argument order change?

Comment (elahrvivaz) [reply geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/parquet/io/SimpleFeatureReadSupport.scala]: i changed it from a Seq to a single type (same with the map converter below), to make it a little clearer what's required. Once the signature had changed, seemed like the binding was 'higher importance' so i put it first... no real reason.

**Pull request merged**